src/Features/CSharp/Portable/EditAndContinue/BreakpointSpans.cs

```diff
@@ -188,6 +188,27 @@ private static int GetEndPosition(SyntaxNodeOrToken nodeOrToken)
                         ? TryCreateSpanForSwitchLabel((SwitchLabelSyntax)node, position)
                         : CreateSpan(caseClause.WhenClause);
 
+                case SyntaxKind.SwitchExpressionArm:
+                    var switchArm = (SwitchExpressionArmSyntax)node;
+                    return CreateSpan((position <= switchArm.WhenClause?.FullSpan.End == true) ? switchArm.WhenClause : (SyntaxNode)switchArm.Expression);
+
+                case SyntaxKind.SwitchExpression when
+                    node is SwitchExpressionSyntax switchExpression &&
+                    switchExpression.Arms.Count > 0 &&
+                    position >= switchExpression.OpenBraceToken.Span.End &&
+                    position <= switchExpression.CloseBraceToken.Span.Start:
+                    // This can occur if the cursor is on a separator. Find the nearest switch arm.
+                    switchArm = switchExpression.Arms[0];
```

Comment:
```suggestion
                        node is SwitchExpressionSyntax switchExpression &&
                        switchExpression.Arms.Count > 0 &&
                        position >= switchExpression.OpenBraceToken.Span.End &&
                        position <= switchExpression.CloseBraceToken.Span.Start:
                        // This can occur if the cursor is on a separator. Find the nearest switch arm.
                        switchArm = switchExpression.Arms[0];
```

Reply:
i'm asking for hte indentation because the above code makes it really hard to separate what's a nested check clause versus what the actual body statements were.